extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeMixPlaylistExtractor.java

```diff
@@ -0,0 +1,188 @@
+    }
+
+    @Nonnull
+    @Override
+    public String getName() throws ParsingException {
+        final String name = playlistData.getString("title");
+        if (name == null) {
+            throw new ParsingException("Could not get playlist name");
+        }
+        return name;
+    }
+
+    @Override
+    public String getThumbnailUrl() throws ParsingException {
+        try {
+            final String playlistId = playlistData.getString("playlistId");
+            try {
+                return getThumbnailUrlFromPlaylistId(playlistId);
+            } catch (ParsingException e) {
+                //fallback to thumbnail of current video. Always the case for channel mix
+                return getThumbnailUrlFromVideoId(
+                    initialData.getObject("currentVideoEndpoint").getObject("watchEndpoint")
+                        .getString("videoId"));
+            }
+        } catch (Exception e) {
+            throw new ParsingException("Could not get playlist thumbnail", e);
+        }
+    }
+
+    @Override
+    public String getBannerUrl() {
+        return "";
+    }
+
+    @Override
+    public String getUploaderUrl() {
+        //Youtube mix are auto-generated
+        return "";
+    }
+
+    @Override
+    public String getUploaderName() {
+        //Youtube mix are auto-generated by YouTube
+        return "YouTube";
+    }
+
+    @Override
+    public String getUploaderAvatarUrl() {
+        //Youtube mix are auto-generated by YouTube
+        return "https://upload.wikimedia.org/wikipedia/commons/thumb/0/09/YouTube_full-color_icon_(2017).svg/320px-YouTube_full-color_icon_(2017).svg.png";
+    }
+
+    @Override
+    public long getStreamCount() {
+        // Auto-generated playlist always start with 25 videos and are endless
+        return ListExtractor.ITEM_COUNT_INFINITE;
+    }
+
+    @Nonnull
+    @Override
+    public InfoItemsPage<StreamInfoItem> getInitialPage() throws ExtractionException {
+        StreamInfoItemsCollector collector = new StreamInfoItemsCollector(getServiceId());
+        collectStreamsFrom(collector, playlistData.getArray("contents"));
+        return new InfoItemsPage<>(collector, getNextPageUrl());
+    }
+
+    @Override
+    public String getNextPageUrl() throws ExtractionException {
+        return getNextPageUrlFrom(playlistData);
+    }
+
+    private String getNextPageUrlFrom(JsonObject playlistData) throws ExtractionException {
+        final JsonObject lastStream = ((JsonObject) playlistData.getArray("contents")
+            .get(playlistData.getArray("contents").size() - 1));
+        if (lastStream == null || lastStream.getObject("playlistPanelVideoRenderer") == null) {
+            throw new ExtractionException("Could not extract next page url");
+        }
+        //Index of video in mix is missing, but adding it doesn't appear to have any effect.
+        //And since the index needs to be tracked by us, it is left out
```

Comment:
I still think `getUrlFromNavigationEndpoint` needs to put the index in the URL. You actually don't need to track it, you just need to do `++navigationEndpoint.getObject("watchEndpoint").getInt("index")`.

Reply:
With tracking i mean the behaviour you get when you play a mix on youtube.com
If you always select the last video in the mix the index jumps from 1 to 25 to 49 to 50 and then stays at 50.
But the index value in watchEndpoint is always 24.

Reply:
I don't think so: 
![yt-mix-json](https://user-images.githubusercontent.com/46277131/77249804-f4916780-6c43-11ea-966c-14476fb58877.png)

Reply:
Is that in the browser?
Since i couldn't get it to simulate that behaviour with the extractor

Reply:
Yes, that's in Firefox. I'll try to reproduce.